internal/bminventory/inventory.go

```diff
@@ -2406,33 +2406,30 @@ func (b *bareMetalInventory) updateNetworks(db *gorm.DB, params installer.Update
 		cluster.ServiceNetworks = params.ClusterUpdateParams.ServiceNetworks
 
 		// TODO MGMT-7365: Deprecate single network
-		if len(params.ClusterUpdateParams.ServiceNetworks) > 0 {
+		if common.IsNetworksSliceNonEmpty(params.ClusterUpdateParams.ServiceNetworks) {
 			updates["service_network_cidr"] = string(params.ClusterUpdateParams.ServiceNetworks[0].Cidr)
 		} else {
 			updates["service_network_cidr"] = ""
 		}
 	}
 
 	if params.ClusterUpdateParams.MachineNetworks != nil {
+		for _, machineNetwork := range params.ClusterUpdateParams.MachineNetworks {
```

Comment:
@YuviGold: I'm moving this validator a bit earlier, i.e. before setting the values for the first time. Let me know if you think this is an issue and if we should really first set and then (re)validate

Reply:
I think that the initial purpose was to verify that the calculated machine cidr is valid. Although in case it wasn't it means we have a bug in our code... So I don't know if that's a requirement. We actually can also validate the machine CIDR during the calculation and assume it is valid (as we don't validate any other value that passes between our own components)

WDYT? @ori-amizur @tsorya

Reply:
I'm not sure I'm not missing something -- machine CIDR is not always calculated by the code; there are cases when user provides it explicitly and then we need to validate it _somewhere_

Reply:
Right.
So either:
* machine CIDR can be calculated and assigned to the cluster; just like when it is provided - it is assigned, and eventually the cluster value is verified. (previous solution)
* It can be calculated and assigned, without the validation (assuming it is valid); only when the value is provided by the user - it is validated before the assignment (your solution)

Reply:
>machine CIDR can be calculated and assigned to the cluster; just like when it is provided - it is assigned, and eventually the cluster value is verified. (previous solution)

I like the fact that we validate values that were calculated, not only those that were coming from the user. But what I don't really like is the order of those actions, i.e.

```
	if params.ClusterUpdateParams.MachineNetworks != nil {
		cluster.MachineNetworks = params.ClusterUpdateParams.MachineNetworks
[...]
	}
 	for index := range cluster.MachineNetworks { 
[...] validate
	}
```

This flow means that we blindy take what came from the user, assign it to the cluster and if it fails later we throw an error, but we don't really rollback the change. I'm not sure however how to make this flow so that all the scenarios are completely covered...

Reply:
There's a rollback. In case of an error - the transaction would be canceled and rolleback.
https://github.com/openshift/assisted-service/blob/51ee4ce418dd208fcc902a4e54064bc74dbbd57b/internal/bminventory/inventory.go#L2067-L2078

But I think we can validate the machine network as well in the calculation itself.
https://github.com/openshift/assisted-service/blob/51ee4ce418dd208fcc902a4e54064bc74dbbd57b/internal/bminventory/inventory.go#L2219-L2239

I'm fine with either way.